packages/admin/admin/src/EditDialog.stackTabsAndGrid.test.tsx

```diff
@@ -0,0 +1,249 @@
+                                                    Add product
+                                                </Button>
+                                            ),
+                                        },
+                                    }}
+                                    disableVirtualization
+                                />
+                            </RouterTab>
+                        </RouterTabs>
+                    </StackPage>
+                    <StackPage name="productEdit" title="Product">
+                        {() => (
+                            <MainContent fullHeight disablePadding>
+                                <SaveBoundary>
+                                    <StackToolbar>
+                                        <ToolbarBackButton />
+                                        <ToolbarAutomaticTitleItem />
+                                        <ToolbarFillSpace />
+                                        <ToolbarActions>
+                                            <SaveBoundarySaveButton />
+                                        </ToolbarActions>
+                                    </StackToolbar>
+                                    <div>Product Edit Page</div>
+                                </SaveBoundary>
+                            </MainContent>
+                        )}
+                    </StackPage>
+                </StackSwitch>
+            </Stack>
+        );
+    }
+
+    it("should navigate to the products page and route when clicking on the products tab", async () => {
+        const history = createMemoryHistory();
+
+        const rendered = render(
+            <Router history={history}>
+                <StackWithGridAndEditDialog />
+            </Router>,
+        );
+
+        expect(history.location.pathname).toBe("/");
+        rendered.getByText("Products").click();
+        expect(screen.getByText("Products")).toBeInTheDocument();
+        expect(history.location.pathname).toBe("/index/products");
+
+        // Check that the Edit Dialog is not open, there was a bug that was fixed
+        // where the edit dialog was open when navigating to a different tab
+        expect(screen.queryByText("Add a new product")).not.toBeInTheDocument();
+    });
+
+    it("should stay on the products page when closing the edit dialog", async () => {
+        const history = createMemoryHistory({
+            initialEntries: ["/", "/products"],
+            initialIndex: 0,
+        });
+
+        const rendered = render(
+            <Router history={history}>
+                <StackWithGridAndEditDialog />
+            </Router>,
+        );
+
+        expect(history.location.pathname).toBe("/");
+        expect(screen.getByText("Customers Page")).toBeInTheDocument();
+
+        rendered.getByText("Products").click();
+        expect(rendered.getByText("Add product")).toBeInTheDocument();
+        expect(history.location.pathname).toBe("/index/products");
```

Comment:
`/index` looks weird to me. Any idea where this comes from?

Reply:
This comes from the SubRouteIndexRoute: https://github.com/vivid-planet/comet/blob/main/packages/admin/admin/src/router/SubRoute.tsx#L18